src/daft-core/src/array/ops/list.rs

```diff
@@ -244,6 +252,128 @@ fn list_sort_helper_fixed_size(
 }
 
 impl ListArray {
+    pub fn value_counts(&self) -> DaftResult<MapArray> {
+        struct IndexRef {
+            index: usize,
+            hash: u64,
+        }
+
+        impl std::hash::Hash for IndexRef {
+            fn hash<H: std::hash::Hasher>(&self, state: &mut H) {
+                self.hash.hash(state);
+            }
+        }
+
+        let original_name = self.name();
+
+        let hashes = self.flat_child.hash(None)?;
+
+        let flat_child = self.flat_child.to_arrow();
+        let flat_child = &*flat_child;
+
+        let is_equal = build_is_equal(
+            flat_child, flat_child, true, // todo: should nans be considered equal?
```

Comment:
What was the verdict on this? NaNs should not be equal, but it would be interesting to allow the user to count the number of NaNs easily.

Reply:
are NaNs considered invalid according to daft? If they are they should be ignored anyway but I agree we should make false probably. 🤔

if they aren't invalid we could get weird behavior. simulating in my head something like 

NaN: 1
NaN: 1

...

etc

for list_values

Reply:
e81e7255a6a3ee5f260fefc7a7ba637165e58914